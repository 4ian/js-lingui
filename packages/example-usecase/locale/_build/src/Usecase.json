{
  "This isn't translated": {
    "origin": [
      [
        "src/Usecase.js",
        28
      ]
    ]
  },
  "msg.label": {
    "defaults": "",
    "origin": [
      [
        "src/Usecase.js",
        32
      ]
    ]
  },
  "Hello World": {
    "origin": [
      [
        "src/Usecase.js",
        36
      ]
    ]
  },
  "My name is {name}": {
    "origin": [
      [
        "src/Usecase.js",
        40
      ]
    ]
  },
  "Read <0>more</0>.": {
    "origin": [
      [
        "src/Usecase.js",
        44
      ]
    ]
  },
  "{genderOfHost, select, female {{numGuests, plural, offset:1 =0 {{host} does not give a party.} =1 {{host} invites {guest} to her party.} =2 {{host} invites {guest} and one other person to her party.} other {{host} invites {guest} and # other people to her party.}}} male {{numGuests, plural, offset:1 =0 {{host} does not give a party.} =1 {{host} invites {guest} to his party.} =2 {{host} invites {guest} and one other person to her party.} other {{host} invites {guest} and # other people to her party.}}} other {{numGuests, plural, offset:1 =0 {{host} does not give a party.} =1 {{host} invites {guest} to their party.} =2 {{host} invites {guest} and one other person to her party.} other {{host} invites {guest} and # other people to her party.}}}}": {
    "origin": [
      [
        "src/Usecase.js",
        50
      ]
    ]
  }
}