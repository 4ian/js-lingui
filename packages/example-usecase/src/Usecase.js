// @flow
import React from 'react'
import { Trans, I18nProvider, Select, Plural } from 'lingui-react'

class NeverUpdate extends React.Component {
  props: {
    children?: any
  }

  shouldComponentUpdate () {
    return false
  }

  render () {
    return <div>{this.props.children}</div>
  }
}

class Usecase extends React.Component {
  props: {
    language: string,
    messages: Object,
    name?: string,
    genderOfHost?: 'male' | 'female',
    numGuests?: number,
    host?: string,
    guest?: string
  }

  render () {
    const {
      language = '', messages = {},
      name = 'Mononoke',
      genderOfHost = 'female',
      numGuests = 4,
      host = 'Wilma',
      guest = 'Fred'
    } = this.props

    return (
      <I18nProvider language={language} messages={messages}>
        <NeverUpdate>
          <ul>
            <li>
              <Trans className="untranslated">{"This isn't translated"}</Trans>
            </li>

            <li>
              <Trans className="customId" id="msg.label" />
            </li>

            <li>
              <Trans className="translated">Hello World</Trans>
            </li>

            <li>
              <Trans className="variable">My name is {name}</Trans>
            </li>

            <li>
              <Trans className="components">
                Read <a href="/mononoke">more</a>.
              </Trans>
            </li>

            <li>
              <Trans className="plural">
                <Select
                  value={genderOfHost}
                  female={
                    <Plural
                      value={numGuests}
                      offset="1"
                      _0={`${host} does not give a party.`}
                      _1={`${host} invites ${guest} to her party.`}
                      _2={`${host} invites ${guest} and one other person to her party.`}
                      other={`${host} invites ${guest} and # other people to her party.`}
                    />
                  }
                  male={
                    <Plural
                      value={numGuests}
                      offset="1"
                      _0={`${host} does not give a party.`}
                      _1={`${host} invites ${guest} to his party.`}
                      _2={`${host} invites ${guest} and one other person to her party.`}
                      other={`${host} invites ${guest} and # other people to her party.`}
                    />
                  }
                  other={
                    <Plural
                      value={numGuests}
                      offset="1"
                      _0={`${host} does not give a party.`}
                      _1={`${host} invites ${guest} to their party.`}
                      _2={`${host} invites ${guest} and one other person to her party.`}
                      other={`${host} invites ${guest} and # other people to her party.`}
                    />
                  }
                />
              </Trans>
            </li>
          </ul>
        </NeverUpdate>
      </I18nProvider>
    )
  }
}

export default Usecase
