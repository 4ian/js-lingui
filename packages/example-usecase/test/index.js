import React from 'react'
import { mount, shallow } from 'enzyme'

import Usecase from '../index'


describe('example-usecase', function() {
  const getText = (element, props = {}) => {
    return mount(<Usecase {...props} />).find(element).text()
  }

  const getHtml = (element, props = {}) => {
    return shallow(<Usecase {...props} />).find(element).html()
  }

  it('should render', function() {
    expect(mount(<Usecase />)).toMatchSnapshot()
  })

  it('should render defaults with warning for untranslated', function() {
    expect(getText('.untranslated')).toEqual("This isn't translated")
  })

  it('should support custom message id', function() {
    expect(getText('.customId')).toEqual('Label')
  })

  it('should render translated string', function() {
    expect(getText('.translated')).toEqual('Ahoj svÄ›te')
  })

  it('should support variable substitution', function() {
    expect(getText('.variable')).toEqual('Jmenuji se Mononoke')
    expect(getText('.variable', { name: 'Fred' })).toEqual('Jmenuji se Fred')
  })

  it('should support nested elements', function() {
    expect(getHtml('.components'))
      .toEqual('<span class=\"components\">Read <a href=\"/mononoke\">more</a>.</span>')
  })

  it('should support pluralization', function() {
    expect(getText('.plural'))
      .toEqual('Wilma invites Fred and 3 other people to her party.')
  })
})
