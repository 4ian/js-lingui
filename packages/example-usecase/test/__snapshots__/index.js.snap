exports[`example-usecase should render 1`] = `
<Usecase>
  <I18nProvider
    language="cs"
    messages={
      Object {
        "Hello World": "Ahoj světe",
        "My name is {name}": "Jmenuji se {name}",
        "msg.label": "Nápis",
      }
    }>
    <NeverUpdate>
      <div>
        <InjectI18n
          className="untranslated"
          id="This isn\'t translated">
          <Trans
            className="untranslated"
            i18n={
              Object {
                "language": "cs",
                "translate": [Function],
              }
            }
            id="This isn\'t translated">
            <span
              className="untranslated">
              This isn\'t translated
            </span>
          </Trans>
        </InjectI18n>
        <InjectI18n
          className="customId"
          defaults=""
          id="msg.label">
          <Trans
            className="customId"
            defaults=""
            i18n={
              Object {
                "language": "cs",
                "translate": [Function],
              }
            }
            id="msg.label">
            <span
              className="customId">
              Nápis
            </span>
          </Trans>
        </InjectI18n>
        <InjectI18n
          className="translated"
          id="Hello World">
          <Trans
            className="translated"
            i18n={
              Object {
                "language": "cs",
                "translate": [Function],
              }
            }
            id="Hello World">
            <span
              className="translated">
              Ahoj světe
            </span>
          </Trans>
        </InjectI18n>
        <InjectI18n
          className="variable"
          id="My name is {name}"
          params={
            Object {
              "name": "Mononoke",
            }
          }>
          <Trans
            className="variable"
            i18n={
              Object {
                "language": "cs",
                "translate": [Function],
              }
            }
            id="My name is {name}"
            params={
              Object {
                "name": "Mononoke",
              }
            }>
            <span
              className="variable">
              Jmenuji se Mononoke
            </span>
          </Trans>
        </InjectI18n>
        <InjectI18n
          className="components"
          components={
            Array [
              <a
                href="/mononoke" />,
            ]
          }
          id="Read <0>more</0>.">
          <Trans
            className="components"
            components={
              Array [
                <a
                  href="/mononoke" />,
              ]
            }
            i18n={
              Object {
                "language": "cs",
                "translate": [Function],
              }
            }
            id="Read <0>more</0>.">
            <span
              className="components">
              Read 
              <a
                href="/mononoke">
                more
              </a>
              .
            </span>
          </Trans>
        </InjectI18n>
        <InjectI18n
          className="plural"
          id="{genderOfHost, select, female {{numGuests, plural, offset:1 =0 {{host} does not give a party.} =1 {{host} invites {guest} to her party.} =2 {{host} invites {guest} and one other person to her party.} other {{host} invites {guest} and # other people to her party.}}} male {{numGuests, plural, offset:1 =0 {{host} does not give a party.} =1 {{host} invites {guest} to his party.} =2 {{host} invites {guest} and one other person to her party.} other {{host} invites {guest} and # other people to her party.}}} other {{numGuests, plural, offset:1 =0 {{host} does not give a party.} =1 {{host} invites {guest} to their party.} =2 {{host} invites {guest} and one other person to her party.} other {{host} invites {guest} and # other people to her party.}}}}"
          params={
            Object {
              "genderOfHost": "female",
              "guest": "Fred",
              "host": "Wilma",
              "numGuests": 4,
            }
          }>
          <Trans
            className="plural"
            i18n={
              Object {
                "language": "cs",
                "translate": [Function],
              }
            }
            id="{genderOfHost, select, female {{numGuests, plural, offset:1 =0 {{host} does not give a party.} =1 {{host} invites {guest} to her party.} =2 {{host} invites {guest} and one other person to her party.} other {{host} invites {guest} and # other people to her party.}}} male {{numGuests, plural, offset:1 =0 {{host} does not give a party.} =1 {{host} invites {guest} to his party.} =2 {{host} invites {guest} and one other person to her party.} other {{host} invites {guest} and # other people to her party.}}} other {{numGuests, plural, offset:1 =0 {{host} does not give a party.} =1 {{host} invites {guest} to their party.} =2 {{host} invites {guest} and one other person to her party.} other {{host} invites {guest} and # other people to her party.}}}}"
            params={
              Object {
                "genderOfHost": "female",
                "guest": "Fred",
                "host": "Wilma",
                "numGuests": 4,
              }
            }>
            <span
              className="plural">
              Wilma invites Fred and 3 other people to her party.
            </span>
          </Trans>
        </InjectI18n>
      </div>
    </NeverUpdate>
  </I18nProvider>
</Usecase>
`;
